plugin/storage/es/factory.go

```diff
@@ -96,13 +101,29 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 	if err != nil {
 		return fmt.Errorf("failed to create primary Elasticsearch client: %w", err)
 	}
-	f.primaryClient = primaryClient
+	f.primaryClient.Store(&primaryClient)
+
+	f.watchers = make([]*fswatcher.FSWatcher, 2)
```

Comment:
Don't preallocate, you don't know if the second watcher will be added. Just use append